static/src/javascripts/projects/common/modules/audio/AudioPlayer.js

```diff
@@ -0,0 +1,641 @@
+        alignItems: 'baseline',
+        flexDirection: 'row',
+        paddingTop: 0,
+        ul: {
+            display: 'flex',
+        },
+        li: {
+            marginLeft: '30px',
+        },
+    },
+    [tablet]: {
+        borderLeft: '1px solid #767676',
+        alignItems: 'baseline',
+        flexDirection: 'row',
+        paddingTop: 0,
+        ul: {
+            display: 'flex',
+        },
+        li: {
+            marginLeft: '30px',
+        },
+    },
+    b: {
+        fontSize: '16px',
+        fontFamily: '"Guardian Egyptian Web",Georgia,serif',
+        fontWeight: '900',
+    },
+    ul: {
+        fontFamily: 'Guardian Text Sans Web',
+        fontSize: '14px',
+    },
+    a: {
+        color: '#cbcbcb', // TODO: add to the palette
+    },
+    li: {
+        marginTop: '12px',
+    },
+    img: {
+        marginRight: '6px',
+        verticalAlign: 'middle',
+        height: '18px',
+    },
+});
+
+const Button = styled('button')(({ isPlay, pillarColor }) => ({
+    background: 'none',
+    border: 0,
+    margin: 0,
+    ':focus': {
+        outline: 'none', // ಠ_ಠ
+    },
+    padding: isPlay ? '0 45px' : 0,
+    svg: {
+        width: isPlay ? '60px' : '26px',
+        height: isPlay ? '60px' : '26px',
+    },
+    circle: {
+        fill: pillarColor,
+        stroke: pillarColor,
+    },
+
+    [leftCol]: {
+        padding: isPlay ? '0 12px' : 0,
+        svg: {
+            width: isPlay ? '50px' : '24px',
+            height: isPlay ? '50px' : '24px',
+        },
+    },
+
+    [wide]: {
+        padding: isPlay ? '0 20px' : 0,
+        svg: {
+            width: isPlay ? '74px' : '26px',
+            height: isPlay ? '74px' : '26px',
+        },
+    },
+}));
+
+type Props = {
+    sourceUrl: string,
+    mediaId: string,
+    downloadUrl: string,
+    iTunesUrl: string,
+    barWidth: number,
+    pillar: string,
+};
+
+type State = {
+    ready: boolean,
+    playing: boolean,
+    currentTime: number,
+    iteration: number,
+    duration: number,
+    volume: number,
+    bins: Array<number>,
+    interval: number,
+    currentOffset: number,
+    canvasH: number,
+    sampler?: ?IntervalID,
+    hasBeenPlayed: boolean,
+};
+
+export default class AudioPlayer extends Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            ready: false,
+            playing: false,
+            currentTime: 0,
+            iteration: 0,
+            duration: NaN,
+            volume: NaN,
+            bins: [],
+            interval: NaN,
+            currentOffset: 0,
+            canvasH: 0,
+            hasBeenPlayed: false,
+        };
+    }
+
+    componentDidMount() {
+        const rect = this.canvas.getBoundingClientRect();
+        this.canvas.setAttribute('viewBox', `0 0 ${rect.width} ${rect.height}`);
+        const nbins = Math.floor(rect.width / 3);
+        this.audio.addEventListener('volumechange', this.onVolumeChange);
```

Comment:
With these addEventListeners will the callbacks not lose the reference to `this`?

If this is an issue then`this.audio.addEventListener('volumechange', this.onVolumeChange).bind(this));` OR
`this.audio.addEventListener('volumechange',() => this.onVolumeChange());` would resolve this.

Reply:
@regiskuckaertz ?

Reply:
They won't, we did exactly what you suggested but inlined in the class definition:
```js
onVolumeChange = () => { /* fat arrow so `this` is preserved */ }
```
instead of
```js
onVolumeChange() { /* `this` not preserved */ }
```